Validate validator schema with schemas definitions

martijnarts opened the pull request:
That's not confusing!

This PR adds a single step of using a JSON Schema that defines JSON Schema schemas to validate the schemas you're trying to validate documents with.

martijnarts committed:
Validate validator schema with schemas definitions

That's not confusing!

Gwildor commented on tests.py:
```diff
@@ -1,3 +1,4 @@
+from collections import namedtuple
```
Missing newline between standard libraries and external libraries.

martijnarts committed:
Reorder imports

martijnarts committed:
Add a single empty line so that @Gwildor won't hate me

dporru committed:
Add test for schema strictness checking

In order to properly do this test, SchemaError is now split up into
several more specific errors. This is all fully backwards compatible.

dporru committed:
Add test for _load_strictness_schema

Added pytest-mock depency for testing.

dporru committed:
Version bump to 0.1.0b2

dporru committed:
Add custom JSONDecode error when not importable

json in python < 3.5 has no JSONDecodeError.

dporru committed:
json < 3.5 gives ValueError

Not TypeError.

dporru committed:
Fix Error enheritance
